tests/variables/test_precipitation.py

```diff
@@ -63,15 +65,15 @@ def test_convert_output(self):
         )
 
     def test_convert_applies_correction_factor(self):
-        result = convert_to_rate(self.precip, self.wspd, self.t)
+        result = precipitation.convert_to_rate_and_correct_undercatch(self.precip, self.wspd, self.t)
         # Ensure precipitation rates are scaled by factor >= 1.02
         self.assertTrue(((result.dropna(dim="time") >= 0).all()).item())
 
     def test_convert_nan_ffill(self):
         precip_nan = xr.DataArray([10.0, 11.0, np.nan, 20.0, 25.0, 29.0, np.nan, 46.0],
                                dims="time",
                                coords=self.precip.coords)
-        result = convert_to_rate(precip_nan, self.wspd, self.t)
+        result = precipitation.convert_to_rate_and_correct_undercatch(precip_nan, self.wspd, self.t)
         # The NaN in precip[2] should be forward-filled before diff
         self.assertFalse(result.isnull()[2].item())
         self.assertFalse(result.isnull()[6].item())
```

Comment:
I have removed this test as it was disfunctional.
NaN in precip should be NaN in precip_rate